crates/sui-types/src/deny_list.rs

```diff
@@ -43,6 +46,73 @@ pub struct PerTypeDenyList {
     pub denied_addresses: Table,
 }
 
+impl DenyList {
+    pub fn check_coin_deny_list(
+        address: SuiAddress,
+        coin_types: BTreeSet<String>,
+        object_store: &dyn ObjectStore,
+    ) -> UserInputResult {
+        let Some(deny_list) = get_coin_deny_list(object_store) else {
+            // TODO: This is where we should fire an invariant violation metric.
+            if cfg!(debug_assertions) {
```

Comment:
Should we do this before landing?

Reply:
I think the risk this firing is really low. It could just be a panic but I was being conservative.